### DISCUSSION TITLE
2022-08-24 : R port detailed overview & next steps

### DISCUSSION THREAD

**Discussion by 7yl4r** [USF-IMARS/chemtax#10] [Meeting Plans, Agendas, Notes]
Chemtax-R code:
* variables need more meaningful names
* scripts need to be organized into a package
* could parallelize to speed things up
* needs more comments

Licensing:
* CHEMTAX already under CC-BY
* suggested to publish v2 under MIT

Hosting:
* can we host the GUI on a shiny server somewhere?
* will we be able to get help with that?

## TODOs:
* (Sebastian) make chemtax-R into R package
* (Tylar) build out GUI framework
* (Simon) send list of people who should be attributed as co-creators
* (Simon) send list of papers that should be included
* (Karen) get DOI for new package
* (Simon & Karen) send email to simulated-annealing-CHEMTAX author

**Comment by sebastiandig** [comment]
Update (Sept 6, 2022):

I have been going through each file and editing variable names for clarity and adding some functionality. 

I would like to know what type of diagnostics would be helpful after running either `randstart`, `bootln`, `bootnp` or `regplot`.  Currently, the only diagnostics that come out are related to overall RMS, or weighted RMS from the `a` matrix (taxa contribution amount) or `b` matrix (pigment ratio matrix). 

ex: 
Random Start Number: 01 of xx

| Iter:  |  RMS:    | Wt RMS:   |  Pigment Ratio RMS:  |   Weighted PR RMS:  |    dRMS: |  dTaxa RMS:  |  dPig RMS: |
| ----- | -------- | ----------- | ----------------------- | ----------------------- | --------- | -------------- | ------------ |
|    0 |   1.11 | 1.14e+03  |              0.000   |            0.00 |    |     |     |
| 5000  | 0.00780 |      3.60 | 0.653 | 11.64  | 1.34e-06 | 5.04e-06 | 2.53e-05


| Terms | Meaning |
| --- | --- |
| RMS |  $=\sqrt{ \frac{(df - a \cdot b)^2}{\text{\\# of pig concentrations overall}}}$ |
| Wt RMS | $=\sqrt{ \frac{ (\frac{(df - a \cdot b)}{sd})^2}{\text{\\# of pig concentrations overall}}}\text{ ;  sd is std. dev of df}$ |
| Pigment Ratio RMS | $=\sqrt{ \frac{(b_{initial} - b_{pred})^2}{\text{\\# of pigments in ratio matrix}}}$  |
| Wt Pigment Ratio RMS | $=\sqrt{ \frac{ (\frac{(b_{initial} - b_{pred})}{sd})^2}{\text{\\# of pigments in ratio matrix}}}\text{ ; sd is std. dev of pigment ratio}$ |
| dRMS | = change in RMS from prev to current, this the the RMS of term 1  and 3 divided by iteration # |
| dTaxa RMS | = change in term 1 RMS  from prev to current divided by iteration #  |
|  dPig RMS | =  change in term 3 RMS  from prev to current divided by iteration # |
